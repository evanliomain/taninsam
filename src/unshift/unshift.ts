/**
 * @module array=>array
 */
/**
 * Return a array with the new elements at the beginning.
 * @param items New elements to add on the Array.
 * @return the function to apply on the array
 * @example
 * ```
 * unshift<number>(4, 5)([1, 2, 3]) // [4, 5, 1, 2, 3]
 * ```
 * @example Using the chain
 * ```
 * chain([1, 2, 3])
 *   .chain(unshift<number>(4, 5))
 *   .value() // [4, 5, 1, 2, 3]
 * ```
 */
export function unshift<T>(
  ...items: T[]
): (array: ReadonlyArray<T>) => ReadonlyArray<T> {
  return (array: ReadonlyArray<T>) => [...items, ...array];
}
