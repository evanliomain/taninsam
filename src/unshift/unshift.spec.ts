import { describe, expect, test } from 'vitest';
import { unshift } from './unshift';

describe('unshift function', () => {
  const input = [1, 2, 3];

  test('is a pure function', () => {
    unshift()(input);
    expect(input).toMatchSnapshot();
  });

  test('[1, 2, 3] |> unshift(4, 5) === [4, 5, 1, 2, 3]', () => {
    expect(unshift(4, 5)(input)).toMatchSnapshot();
  });

  test('[] |> unshift(4, 5) === [4, 5]', () => {
    expect(unshift(4, 5)([])).toMatchSnapshot();
  });
});
