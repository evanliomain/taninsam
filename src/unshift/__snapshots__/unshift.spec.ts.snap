// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unshift function [] |> unshift(4, 5) === [4, 5] 1`] = `
Array [
  4,
  5,
]
`;

exports[`unshift function [1, 2, 3] |> unshift(4, 5) === [4, 5, 1, 2, 3] 1`] = `
Array [
  4,
  5,
  1,
  2,
  3,
]
`;

exports[`unshift function is a pure function 1`] = `
Array [
  1,
  2,
  3,
]
`;
